src/TurbulenceClosures/turbulence_closure_implementations/convective_adjustment_vertical_diffusivity.jl

```diff
@@ -0,0 +1,220 @@
+
+* `background_κz` : Vertical tracer diffusivity in regions with stable buoyancy gradients.
+
+* `convective_νz` : Vertical viscosity in regions with unstable buoyancy gradients. Either
+                    a number, function, array, or field.
+
+* `background_κz` : Vertical viscosity in regions with stable buoyancy gradients.
+
+* `time_discretization` : Either `ExplicitTimeDiscretization` or `VerticallyImplicitTimeDiscretization`.
+"""
+function ConvectiveAdjustmentVerticalDiffusivity(FT=Float64;
+                                                 convective_κz = zero(FT),
+                                                 convective_νz = zero(FT),
+                                                 background_κz = zero(FT),
+                                                 background_νz = zero(FT),
+                                                 time_discretization::TD = VerticallyImplicitTimeDiscretization()) where TD
+
+    return ConvectiveAdjustmentVerticalDiffusivity{TD}(convective_κz, convective_νz,
+                                                       background_κz, background_νz)
+end
+
+const CAVD = ConvectiveAdjustmentVerticalDiffusivity
+
+#####
+##### Diffusivity field utilities
+#####
+
+function with_tracers(tracers, closure::ConvectiveAdjustmentVerticalDiffusivity{TD}) where TD
+    background_κz = tracer_diffusivities(tracers, closure.background_κz)
+    convective_κz = tracer_diffusivities(tracers, closure.convective_κz)
+    return ConvectiveAdjustmentVerticalDiffusivity{TD}(convective_κz,
+                                                       closure.convective_νz,
+                                                       background_κz,
+                                                       closure.background_νz)
+end
+
+
+function DiffusivityFields(arch, grid, tracer_names, bcs, closure::CAVD)
+    data = new_data(Bool, arch, grid, (Center, Center, Face))
+    stable_buoyancy_gradient = Field(Center, Center, Face, arch, grid, nothing, data)
+    return (; stable_buoyancy_gradient)
+end       
+
+function calculate_diffusivities!(diffusivities, arch, grid, closure::CAVD, buoyancy, velocities, tracers)
+
+    event = launch!(arch, grid, :xyz,
+                    compute_stability!, diffusivities, grid, tracers, buoyancy,
+                    dependencies = device_event(arch))
+
+    wait(device(arch), event)
+
+    return nothing
+end
+
+@inline is_stableᶜᶜᶠ(i, j, k, grid, tracers, buoyancy) = ∂z_b(i, j, k, grid, buoyancy, tracers) > 0
+
+@kernel function compute_stability!(diffusivities, grid, tracers, buoyancy)
+    i, j, k, = @index(Global, NTuple)
+    @inbounds  diffusivities.stable_buoyancy_gradient[i, j, k] = is_stableᶜᶜᶠ(i, j, k, grid, tracers, buoyancy)
+end
+
+#####
+##### Fluxes
+#####
+
+# u
+@inline function νᶠᶜᶠ(i, j, k, grid, clock, closure::CAVD, stable_buoyancy_gradient)
+    @inbounds stableᶠᶜᶠ = stable_buoyancy_gradient[i, j, k] || stable_buoyancy_gradient[i+1, j, k]
```

Comment:
@glwagner is ```i+1``` correct? Isn't ```u[i,j,k]``` bracketed by ```rho[i-1,j,k]``` and ```rho[i,j,k]```?

Reply:
quite, good eye 👁️  !